### PR TITLE
Olofk/director

### PR DESCRIPTION
**Pull request by olofkallander**

### CODE REVIEW

File: bridge/engine/EngineStreamDirector.h [hunk 1]
```diff
@@ -268,58 +276,62 @@ class EngineStreamDirector
...
                 uplinkEstimateKbps,
                 endpointIdHash,
-                participantStream.desiredHighestEstimatedPinnedLevel,
-                participantStream.highestEstimatedPinnedLevel);
+                participantStream.pinQualityLevel,
+                participantStream.desiredPinQualityLevel,
+                participantStream.unpinQualityLevel);
 
-            participantStream.highestEstimatedPinnedLevel = participantStream.desiredHighestEstimatedPinnedLevel;
+            participantStream.pinQualityLevel = participantStream.desiredPinQualityLevel;
```
Comment (olofkallander) [inline bridge/engine/EngineStreamDirector.h:317]: here we actually change the quality setting

File: bridge/engine/EngineStreamDirector.h [hunk 2]
```diff
@@ -268,58 +276,62 @@ class EngineStreamDirector
...
```
Comment (olofkallander) [inline bridge/engine/EngineStreamDirector.h:300]: Actually set quality to use

File: bridge/engine/EngineStreamDirector.h [hunk 3]
```diff
@@ -460,30 +472,29 @@ class EngineStreamDirector
 
         const auto pinMapItr = _pinMap.find(toEndpointIdHash);
         const bool fromPinnedEndpoint = pinMapItr != _pinMap.end() && isSsrcFromParticipant(pinMapItr->second, ssrc);
-        const auto maxWantedQuality =
-            (fromPinnedEndpoint ? viewer->desiredHighestEstimatedPinnedLevel : viewer->desiredUnpinnedLevel);
+        const auto assignedQuality = (fromPinnedEndpoint ? viewer->pinQualityLevel : viewer->unpinQualityLevel);
```
Comment (olofkallander) [inline bridge/engine/EngineStreamDirector.h:475]: This was the main bug. We used desired quality to figure out if ssrc is used, not the actual selected quality

File: bridge/engine/EngineStreamDirector.h [hunk 4]
```diff
@@ -902,24 +904,24 @@ class EngineStreamDirector
...
+            if (pinnedBy == senderEndpointIdHash || pinTarget != senderEndpointIdHash)
             {
                 continue;
             }
 
             const auto& participant = _participantStreams.find(pinnedBy);
 
             if (participant != _participantStreams.end() &&
-                quality == std::min(participant->second.desiredHighestEstimatedPinnedLevel, highestAvailableQuality))
+                quality == std::min(participant->second.pinQualityLevel, highestAvailableQuality))
```
Comment (olofkallander) [inline bridge/engine/EngineStreamDirector.h:916]: This was also bug that it reported an ssrc to be in use before the quality upgrade has happened. And thus another ssrc reported as not being used.

**Review by RicardoMDomingues — APPROVED** [review]

**Pull request merged**